sql/execution_pipeline.h

```diff
@@ -331,6 +346,20 @@ struct PipelineElement: public std::enable_shared_from_this<PipelineElement> {
          std::shared_ptr<SqlExpression> where = SqlExpression::TRUE,
          OrderByExpression orderBy = OrderByExpression());
     
+    /** Add a join to the pipeline that is pre-bound into the immediate outer
+        scope.
+    */
+    std::shared_ptr<PipelineElement>
+    join(std::shared_ptr<TableExpression> left,
```

Comment:
It seems that we use either the table expression or the provided "pre-bound" expression.  Why don't we have two overloaded functions (here and above) instead?